cycode/cli/code_scanner.py

```diff
@@ -336,7 +337,19 @@ def scan_disk_files(context: click.Context, path: str, files_to_scan: List[str])
         if not content:
             continue
 
-        documents.append(Document(file, content, is_git_diff))
+        file_name = file
+
+        if _is_iac(scan_type) and _is_tfplan_json_file(file, content):
+            try:
+                content = tf_content_generator.generate_tf_content_from_tfplan(content)
+            except KeyError:
+                _handle_exception(
```

Comment:
do we want to soft fail and exit when failing to generate tfplan?

Reply:
@MichalBor I think that yes, the use case here is that the user will want to run scan over tf plan during pipeline so the expectation is that we will block the pipeline if scan didnt pass successfull.
such case is end case because tf plan is something the automatically generated but if it will happen I believe that it's better to throw error instead os swallow the exception.

Reply:
looks like it will print the error immediately and will break the output of the progress bar

Reply:
fixed in https://github.com/cycodehq-public/cycode-cli/pull/153/commits/7c3296a9867901c3b40a32c1727a620dbbf746be not it doesn't break progress bar